crates/hyperqueue/src/worker/start.rs

```diff
@@ -95,14 +95,14 @@ impl TaskLauncher for HqTaskLauncher {
             let body: TaskBody = tako::transfer::auth::deserialize(&task.body)?;
             let TaskBody {
                 mut program,
-                pin,
+                pin: pin_mode,
                 task_dir,
                 job_id,
                 task_id,
             } = body;
 
-            if pin {
-                pin_program(&mut program, allocation);
+            if let Some(pin_mode) = pin_mode {
+                pin_program(&mut program, allocation, pin_mode);
                 program.env.insert(HQ_PIN.into(), "1".into());
```

Comment:
Maybe storing a name of the mode into HQ_PIN variable?

Reply:
Good idea.